### DISCUSSION TITLE
Access to integration and communication intervals value from a model

### DISCUSSION THREAD

**Discussion by mpradovelasco** [OpenModelica/OpenModelica#11209] [Q&A] [answered]
Hi, 

After obtaining a simulation error due to use:
`delay(xsf[i],smallTime)` where smallTime was 100*Modelica.Constants.small (I may feel the trouble in the dramatic small value of smallTime; of course I tested as delay(xsf[i], 1 ) successfully), I wonder if I have access to the true integration interval from the model (even when it is variable), in such a way that I could take a proper multiple of it (something like `delay(xsf[i],k*INT, delayMax` ), where INT is the integration interval.

I wonder also if I have access to the communication interval from the model. Although it may be defined in the annotation - experiment clause: experiment(Interval = 60), I do not know how can access to the Interval variable. 

Thank you in advance!

**Comment by casella** [comment, marked as answer]
If you want to see the actual time steps, deselect Simulation Setup | Output | Equidistant Time Grid. This way, the results point you see are the actual accepted ODE integrator steps.

The communication interval is the one you set with the Interval annotation. You can also see that because the result points in the .mat file are multiples of it, although there can be extra points corresponding to events if Store Variables At Events was selected.

**Reply by mpradovelasco** [reply]
Thank you for your answer!
